### PR TITLE
test: integration tests for `/me`

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

test: integration tests for `/me`

## Testing

None.

### CODE REVIEW

File: packages/integration-tests/tests/me.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { generatePassword, generateUsername } from '@/utils';
+
+const createUser = async () => {
+  const username = generateUsername();
+
+  return authedAdminApi
+    .post('users', {
+      json: {
+        username,
+        password: generatePassword(),
```
Comment (IceHe) [inline packages/integration-tests/tests/me.test.ts:13]: better inline both `username` and `password`?

File: packages/integration-tests/tests/me.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        headers: {
+          'development-user-id': user.id,
+        },
+      })
+      .json<ArbitraryObject>();
+
+    expect(customData).toEqual({});
+  });
+
+  it('should update user custom data', async () => {
```
Comment (IceHe) [inline packages/integration-tests/tests/me.test.ts:34]: nit: maybe better merge these two cases?

File: packages/integration-tests/tests/me.test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    const user = await createUser();
+
+    const newCustomData = {
+      foo: 'bar',
+    };
+
+    await api.patch('me/custom-data', {
+      headers: {
```
Comment (gao-sun) [inline packages/integration-tests/tests/me.test.ts:42]: 2 headers can be shared

File: packages/integration-tests/tests/me.test.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/integration-tests/tests/me.test.ts:13]: I prefer to set the `username` and `name` with the same value, so the username is not inline.

File: packages/integration-tests/tests/me.test.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/integration-tests/tests/me.test.ts:34]: Just want to keep the test more focused on their test targets.

File: packages/integration-tests/tests/me.test.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      })
+      .json<ArbitraryObject>();
+
+    expect(customData).toEqual({});
+  });
+
+  it('should update user custom data successfully', async () => {
+    const user = await createUser();
+
+    const developmentUserHeaders = {
```
Comment (gao-sun) [inline packages/integration-tests/tests/me.test.ts:48]: i think you can just use 

```ts
const headers = Object.freeze();
```

And free to use the const directly

File: packages/integration-tests/tests/me.test.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+describe('api `/me`', () => {
+  it('should get user info successfully', async () => {
+    const user = await createUser();
+    const userInfo = await api
+      .get(`me`, { headers: { 'development-user-id': user.id } })
+      .json<UserInfo>();
+
+    for (const field of userInfoSelectFields) {
```
Comment (gao-sun) [inline packages/integration-tests/tests/me.test.ts:27]: should test id equality

**Review by gao-sun — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]

**Pull request merged**